File: algorithms/1_easy/M/01189_maximum_number_of_balloons/freqs_arr.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        freqs = [0] * 26

        for ch in text:
            freqs[ord(ch) - ord('a')] += 1

        return min([freqs[ord('b') - ord('a')], \
                    freqs[ord('a') - ord('a')], \
                    freqs[ord('l') - ord('a')] // 2, \
                    freqs[ord('o') - ord('a')] // 2, \
                    freqs[ord('n') - ord('a')]])

File: algorithms/1_easy/M/01189_maximum_number_of_balloons/freqs_arr.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {

        int freqs[26]{};

        for (const char &ch: text) {
            freqs[ch - 'a'] += 1;
        }

        return min({freqs['b' - 'a'], freqs['a' - 'a'], freqs['l' - 'a'] / 2, freqs['o' - 'a'] / 2, freqs['n' - 'a']});
    }
};
